### PR TITLE
Implement 3XDAMVDEM Logic for Arrows

### PR DESCRIPTION
**Pull request by kphoenix137**
The Deadly Hunter bow contains the 3XDAMVDEM affix power which isn't present in missiles.cpp, preventing the affix from doing anything altogether. This fix adds the logic from player.cpp in a consistent way with how damage gets handled in melee combat.

### CODE REVIEW

File: Source/missiles.cpp [hunk 1]
```diff
@@ -275,6 +275,10 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 	if (resist)
 		dam >>= 2;
 
+	if (monster.MData->mMonstClass == MonsterClass::Demon && (player._pIFlags & ISPL_3XDAMVDEM) != 0) {
+			dam *= 3;
```
Comment (AJenbo) [inline Source/missiles.cpp:279]:
```suggestion
		dam *= 3;
```

**Comment by ikonomov** [comment]
This is clearly a bug, but if this bow was doing three times the total char damage it will be far too powerful, totally breaking the balance in the game.  I propose another fix, to simply remove the IPL_3XDAMVDEM affix from its properties and assume that the bug is not implementing the damage, but removing the affix from the item once they decided that the extra damage should not be implemented.

**Comment by kphoenix137** [comment]
> This is clearly a bug, but if this bow was doing three times the total char damage it will be far too powerful, totally breaking the balance in the game.  I propose another fix, to simply remove the IPL_3XDAMVDEM affix from its properties and assume that the bug is not implementing the damage, but removing the affix from the item once they decided that the extra damage should not be implemented.

The scope of DevilutionX isn't to change the balance of the game, but to fix but and improve the game engine

**Comment by ikonomov** [comment]
So let's assume the bug is having the affix.

**Comment by ikonomov** [comment]
I want Deadly Hunter's bug to be fixed, because it should be fixed.  I just pointed out that there are 2 ways to fix it, and one doesn't involve breaking the balance, while the other one does, so when deciding which way to fix it, it might be worth considering.

**Review by AJenbo — DISMISSED** [review]

**Comment by StephenCWills** [comment]
> Afaik this will cause all missiles shot by the player 3x, it should be limited to only arrows.

The `MissilesData[t].mType == 0` condition should only apply to arrows so that should already be handled.

**Review by AJenbo — APPROVED** [review]

**Pull request merged**